lib/search/grouped_search_results.rb

```diff
@@ -101,6 +105,15 @@ def add(object)
       else
         (self.public_send(type)) << object
       end
+
+      case type
+      when "posts"
+        add_topic_category(object.topic.category&.parent_category)
+        add_topic_category(object.topic.category)
+      when "topics"
+        add_topic_category(object.category&.parent_category)
+        add_topic_category(object.category)
+      end
```

Comment:
I think we can simplify this if we just add a `category` delegate method to the `Post` model such that an instance of `Post` will respond to `category` as well. 

```
add_topic_category(object.category) if object.responds_to?(:category)
```